build failure: musl

When building against musl, i found the following build failure:

<details><summary>Build log</summary>
<p>

```
Making all in auplugin
make[2]: Entering directory '/home/grimmauld/coding/audit-userspace/auplugin'
Making all in test
make[3]: Entering directory '/home/grimmauld/coding/audit-userspace/auplugin/test'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/grimmauld/coding/audit-userspace/auplugin/test'
make[3]: Entering directory '/home/grimmauld/coding/audit-userspace/auplugin'
depbase=`echo auplugin.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/nix/store/8hgbqvf9g6pnwyrq9kacgyilgbnfnqix-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I.. -I../lib -I../common -I../auparse -I../auplugin -I../audisp -I../src  -fPIC -DPIC -D_GNU_SOURCE -Wno-format-truncation -Wno-unused-but-set-variable -g -O2 -MT auplugin.lo -MD -MP -MF $depbase.Tpo -c -o auplugin.lo auplugin.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../lib -I../common -I../auparse -I../auplugin -I../audisp -I../src -fPIC -DPIC -D_GNU_SOURCE -Wno-format-truncation -Wno-unused-but-set-variable -g -O2 -MT auplugin.lo -MD -MP -MF .deps/auplugin.Tpo -c auplugin.c  -fPIC -DPIC -o .libs/auplugin.o
In file included from ../lib/libaudit.h:52,
                 from ../audisp/libdisp.h:5,
                 from auplugin.c:33:
../lib/audit_logging.h: In function 'audit_encode_nv_string':
../lib/audit_logging.h:63:9: error: expected declaration specifiers before '__attr_dealloc_free'
   63 |         __attr_dealloc_free;
      |         ^~~~~~~~~~~~~~~~~~~
../lib/libaudit.h:109:1: warning: empty declaration
  109 | struct audit_sig_info {
      | ^~~~~~
../lib/libaudit.h:117:1: warning: empty declaration
  117 | struct audit_message {
      | ^~~~~~
../lib/libaudit.h:123:1: warning: empty declaration
  123 | struct daemon_conf;
      | ^~~~~~
../lib/libaudit.h:125:1: warning: empty declaration
  125 | struct audit_reply {
      | ^~~~~~
../lib/libaudit.h:159:1: warning: empty declaration
  159 | struct audit_dispatcher_header {
      | ^~~~~~
../lib/libaudit.h:194:3: error: storage class specified for parameter 'machine_t'
  194 | } machine_t;
      |   ^~~~~~~~~
../lib/libaudit.h:201:3: error: storage class specified for parameter 'auditfail_t'
  201 | } auditfail_t;
      |   ^~~~~~~~~~~
../lib/libaudit.h:204:52: error: storage class specified for parameter 'message_t'
  204 | typedef enum { MSG_STDERR, MSG_SYSLOG, MSG_QUIET } message_t;
      |                                                    ^~~~~~~~~
../lib/libaudit.h:205:34: error: storage class specified for parameter 'debug_message_t'
  205 | typedef enum { DBG_NO, DBG_YES } debug_message_t;
      |                                  ^~~~~~~~~~~~~~~
../lib/libaudit.h:206:25: error: expected declaration specifiers or '...' before 'message_t'
  206 | void set_aumessage_mode(message_t mode, debug_message_t debug);
      |                         ^~~~~~~~~
../lib/libaudit.h:206:41: error: expected declaration specifiers or '...' before 'debug_message_t'
  206 | void set_aumessage_mode(message_t mode, debug_message_t debug);
      |                                         ^~~~~~~~~~~~~~~
../lib/libaudit.h:209:62: error: storage class specified for parameter 'reply_t'
  209 | typedef enum { GET_REPLY_BLOCKING=0, GET_REPLY_NONBLOCKING } reply_t;
      |                                                              ^~~~~~~
../lib/libaudit.h:210:55: error: expected declaration specifiers or '...' before 'reply_t'
  210 | int  audit_get_reply(int fd, struct audit_reply *rep, reply_t block,
      |                                                       ^~~~~~~
../lib/libaudit.h:250:26: error: expected declaration specifiers or '...' before 'auditfail_t'
  250 | int get_auditfail_action(auditfail_t *failmode);
      |                          ^~~~~~~~~~~
../lib/libaudit.h:255:36: error: storage class specified for parameter 'rep_wait_t'
  255 | typedef enum { WAIT_NO, WAIT_YES } rep_wait_t;
      |                                    ^~~~~~~~~~
../lib/libaudit.h:256:42: error: expected declaration specifiers or '...' before 'rep_wait_t'
  256 | int  audit_set_pid(int fd, uint32_t pid, rep_wait_t wmode) __wur;
      |                                          ^~~~~~~~~~
In file included from ../audisp/libdisp.h:6:
../src/auditd-config.h:34:45: error: storage class specified for parameter 'daemon_t'
   34 | typedef enum { D_FOREGROUND, D_BACKGROUND } daemon_t;
      |                                             ^~~~~~~~
../src/auditd-config.h:35:48: error: storage class specified for parameter 'logging_formats'
   35 | typedef enum { LF_RAW, LF_NOLOG, LF_ENRICHED } logging_formats;
      |                                                ^~~~~~~~~~~~~~~
../src/auditd-config.h:36:82: error: storage class specified for parameter 'flush_technique'
   36 | typedef enum { FT_NONE, FT_INCREMENTAL, FT_INCREMENTAL_ASYNC, FT_DATA, FT_SYNC } flush_technique;
      |                                                                                  ^~~~~~~~~~~~~~~
../src/auditd-config.h:38:38: error: storage class specified for parameter 'failure_action_t'
   38 |                 FA_SINGLE, FA_HALT } failure_action_t;
      |                                      ^~~~~~~~~~~~~~~~
../src/auditd-config.h:40:31: error: storage class specified for parameter 'size_action'
   40 |                SZ_KEEP_LOGS } size_action;
      |                               ^~~~~~~~~~~
../src/auditd-config.h:41:43: error: storage class specified for parameter 'log_test_t'
   41 | typedef enum { TEST_AUDITD, TEST_SEARCH } log_test_t;
      |                                           ^~~~~~~~~~
../src/auditd-config.h:42:63: error: storage class specified for parameter 'node_t'
   42 | typedef enum { N_NONE, N_HOSTNAME, N_FQD, N_NUMERIC, N_USER } node_t;
      |                                                               ^~~~~~
../src/auditd-config.h:44:26: error: storage class specified for parameter 'overflow_action_t'
   44 |                 O_HALT } overflow_action_t;
      |                          ^~~~~~~~~~~~~~~~~
../src/auditd-config.h:45:50: error: storage class specified for parameter 'transport_t'
   45 | typedef enum { T_TCP, T_TLS, T_KRB5, T_LABELED } transport_t;
      |                                                  ^~~~~~~~~~~
../src/auditd-config.h:49:9: error: expected specifier-qualifier-list before 'daemon_t'
   49 |         daemon_t daemonize;
      |         ^~~~~~~~
../src/auditd-config.h:47:1: warning: empty declaration
   47 | struct daemon_conf
      | ^~~~~~
../src/auditd-config.h:107:45: error: expected declaration specifiers or '...' before 'log_test_t'
  107 | int load_config(struct daemon_conf *config, log_test_t lt);
      |                                             ^~~~~~~~~~
../audisp/libdisp.h:12:3: error: storage class specified for parameter 'event_t'
   12 | } event_t;
      |   ^~~~~~~
../audisp/libdisp.h:19:21: error: expected declaration specifiers or '...' before 'event_t'
   19 | int libdisp_enqueue(event_t *e);
      |                     ^~~~~~~
In file included from /nix/store/3hmxj8k54zn0x0nn9jh67c84hc55bfnl-musl-1.2.5-dev/include/stddef.h:19,
                 from auplugin.h:27,
                 from auplugin.c:35:
/nix/store/3hmxj8k54zn0x0nn9jh67c84hc55bfnl-musl-1.2.5-dev/include/bits/alltypes.h:41:54: error: storage class specified for parameter 'max_align_t'
   41 | typedef struct { long long __ll; long double __ld; } max_align_t;
      |                                                      ^~~~~~~~~~~
/nix/store/3hmxj8k54zn0x0nn9jh67c84hc55bfnl-musl-1.2.5-dev/include/bits/alltypes.h:60:15: error: storage class specified for parameter 'ptrdiff_t'
   60 | typedef _Addr ptrdiff_t;
      |               ^~~~~~~~~
In file included from ../auparse/auparse.h:27,
                 from auplugin.h:29:
../auparse/auparse-defs.h:38:69: error: storage class specified for parameter 'ausource_t'
   38 |         AUSOURCE_DESCRIPTOR, AUSOURCE_FILE_POINTER, AUSOURCE_FEED } ausource_t;
      |                                                                     ^~~~~~~~~~
../auparse/auparse-defs.h:49:3: error: storage class specified for parameter 'ausearch_op_t'
   49 | } ausearch_op_t;
      |   ^~~~~~~~~~~~~
../auparse/auparse-defs.h:53:31: error: storage class specified for parameter 'austop_t'
   53 |         AUSEARCH_STOP_FIELD } austop_t;
      |                               ^~~~~~~~
../auparse/auparse-defs.h:58:50: error: storage class specified for parameter 'ausearch_rule_t'
   58 |         AUSEARCH_RULE_AND, AUSEARCH_RULE_REGEX } ausearch_rule_t;
      |                                                  ^~~~~~~~~~~~~~~
../auparse/auparse-defs.h:67:3: error: storage class specified for parameter 'au_event_t'
   67 | } au_event_t;
      |   ^~~~~~~~~~
../auparse/auparse-defs.h:71:39: error: storage class specified for parameter 'auparse_cb_event_t'
   71 | typedef enum {AUPARSE_CB_EVENT_READY} auparse_cb_event_t;
      |                                       ^~~~~~~~~~~~~~~~~~
../auparse/auparse-defs.h:93:3: error: storage class specified for parameter 'auparse_type_t'
   93 | } auparse_type_t;
      |   ^~~~~~~~~~~~~~
../auparse/auparse-defs.h:97:35: error: storage class specified for parameter 'auparse_esc_t'
   97 |         AUPARSE_ESC_SHELL_QUOTE } auparse_esc_t;
      |                                   ^~~~~~~~~~~~~
../auparse/auparse-defs.h:101:34: error: storage class specified for parameter 'auparse_destroy_what_t'
  101 |         AUPARSE_DESTROY_COMMON } auparse_destroy_what_t;
      |                                  ^~~~~~~~~~~~~~~~~~~~~~
../auparse/auparse-defs.h:104:49: error: storage class specified for parameter 'normalize_option_t'
  104 | typedef enum { NORM_OPT_ALL, NORM_OPT_NO_ATTRS} normalize_option_t;
      |                                                 ^~~~~~~~~~~~~~~~~~
../auparse/auparse.h:48:23: error: storage class specified for parameter 'auparse_state_t'
   48 | typedef struct opaque auparse_state_t;
      |                       ^~~~~~~~~~~~~~~
../auparse/auparse.h:50:16: error: storage class specified for parameter 'user_destroy'
   50 | typedef void (*user_destroy)(void *user_data);
      |                ^~~~~~~~~~~~
../auparse/auparse.h:51:38: error: expected declaration specifiers or '...' before 'auparse_state_t'
   51 | typedef void (*auparse_callback_ptr)(auparse_state_t *au,
      |                                      ^~~~~~~~~~~~~~~
../auparse/auparse.h:52:25: error: expected declaration specifiers or '...' before 'auparse_cb_event_t'
   52 |                         auparse_cb_event_t cb_event_type, void *user_data);
      |                         ^~~~~~~~~~~~~~~~~~
../auparse/auparse.h:55:22: error: expected declaration specifiers or '...' before 'auparse_state_t'
   55 | void auparse_destroy(auparse_state_t *au);
      |                      ^~~~~~~~~~~~~~~
../auparse/auparse.h:56:26: error: expected declaration specifiers or '...' before 'auparse_state_t'
   56 | void auparse_destroy_ext(auparse_state_t *au, auparse_destroy_what_t what);
      |                          ^~~~~~~~~~~~~~~
../auparse/auparse.h:56:47: error: expected declaration specifiers or '...' before 'auparse_destroy_what_t'
   56 | void auparse_destroy_ext(auparse_state_t *au, auparse_destroy_what_t what);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~
../auparse/auparse.h:57:1: error: expected declaration specifiers before 'auparse_state_t'
   57 | auparse_state_t *auparse_init(ausource_t source, const void *b)
      | ^~~~~~~~~~~~~~~
../auparse/auparse.h:59:24: error: expected declaration specifiers or '...' before 'auparse_state_t'
   59 | int auparse_new_buffer(auparse_state_t *au, const char *data, size_t data_len)
      |                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:61:18: error: expected declaration specifiers or '...' before 'auparse_state_t'
   61 | int auparse_feed(auparse_state_t *au, const char *data, size_t data_len)
      |                  ^~~~~~~~~~~~~~~
../auparse/auparse.h:63:30: error: expected declaration specifiers or '...' before 'auparse_state_t'
   63 | void auparse_feed_age_events(auparse_state_t *au);
      |                              ^~~~~~~~~~~~~~~
../auparse/auparse.h:64:24: error: expected declaration specifiers or '...' before 'auparse_state_t'
   64 | int auparse_flush_feed(auparse_state_t *au);
      |                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:65:33: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
   65 | int auparse_feed_has_data(const auparse_state_t *au);
      |                                 ^~~~~~~~~~~~~~~
../auparse/auparse.h:65:49: error: expected ';', ',' or ')' before '*' token
   65 | int auparse_feed_has_data(const auparse_state_t *au);
      |                                                 ^
../auparse/auparse.h:66:34: error: expected declaration specifiers or '...' before 'auparse_state_t'
   66 | int auparse_feed_has_ready_event(auparse_state_t *au);
      |                                  ^~~~~~~~~~~~~~~
../auparse/auparse.h:67:27: error: expected declaration specifiers or '...' before 'auparse_state_t'
   67 | void auparse_add_callback(auparse_state_t *au, auparse_callback_ptr callback,
      |                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:67:48: error: unknown type name 'auparse_callback_ptr'
   67 | void auparse_add_callback(auparse_state_t *au, auparse_callback_ptr callback,
      |                                                ^~~~~~~~~~~~~~~~~~~~
../auparse/auparse.h:68:42: error: expected declaration specifiers or '...' before 'user_destroy'
   68 |                         void *user_data, user_destroy user_destroy_func);
      |                                          ^~~~~~~~~~~~
../auparse/auparse.h:69:30: error: expected declaration specifiers or '...' before 'auparse_state_t'
   69 | void auparse_set_escape_mode(auparse_state_t *au, auparse_esc_t mode);
      |                              ^~~~~~~~~~~~~~~
../auparse/auparse.h:69:51: error: expected declaration specifiers or '...' before 'auparse_esc_t'
   69 | void auparse_set_escape_mode(auparse_state_t *au, auparse_esc_t mode);
      |                                                   ^~~~~~~~~~~~~
../auparse/auparse.h:70:19: error: expected declaration specifiers or '...' before 'auparse_state_t'
   70 | int auparse_reset(auparse_state_t *au);
      |                   ^~~~~~~~~~~~~~~
../auparse/auparse.h:71:29: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
   71 | char *auparse_metrics(const auparse_state_t *au)
      |                             ^~~~~~~~~~~~~~~
../auparse/auparse.h:71:45: error: expected ';', ',' or ')' before '*' token
   71 | char *auparse_metrics(const auparse_state_t *au)
      |                                             ^
../auparse/auparse.h:75:29: error: expected declaration specifiers or '...' before 'auparse_state_t'
   75 | int ausearch_add_expression(auparse_state_t *au, const char *expression,
      |                             ^~~~~~~~~~~~~~~
../auparse/auparse.h:76:43: error: expected declaration specifiers or '...' before 'ausearch_rule_t'
   76 |                             char **error, ausearch_rule_t how);
      |                                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:77:23: error: expected declaration specifiers or '...' before 'auparse_state_t'
   77 | int ausearch_add_item(auparse_state_t *au, const char *field, const char *op,
      |                       ^~~~~~~~~~~~~~~
../auparse/auparse.h:78:44: error: expected declaration specifiers or '...' before 'ausearch_rule_t'
   78 |                         const char *value, ausearch_rule_t how);
      |                                            ^~~~~~~~~~~~~~~
../auparse/auparse.h:79:35: error: expected declaration specifiers or '...' before 'auparse_state_t'
   79 | int ausearch_add_interpreted_item(auparse_state_t *au, const char *field,
      |                                   ^~~~~~~~~~~~~~~
../auparse/auparse.h:80:60: error: expected declaration specifiers or '...' before 'ausearch_rule_t'
   80 |                         const char *op, const char *value, ausearch_rule_t how);
      |                                                            ^~~~~~~~~~~~~~~
../auparse/auparse.h:81:33: error: expected declaration specifiers or '...' before 'auparse_state_t'
   81 | int ausearch_add_timestamp_item(auparse_state_t *au, const char *op, time_t sec,
      |                                 ^~~~~~~~~~~~~~~
../auparse/auparse.h:82:49: error: expected declaration specifiers or '...' before 'ausearch_rule_t'
   82 |                                 unsigned milli, ausearch_rule_t how);
      |                                                 ^~~~~~~~~~~~~~~
../auparse/auparse.h:83:36: error: expected declaration specifiers or '...' before 'auparse_state_t'
   83 | int ausearch_add_timestamp_item_ex(auparse_state_t *au, const char *op,
      |                                    ^~~~~~~~~~~~~~~
../auparse/auparse.h:84:54: error: expected declaration specifiers or '...' before 'ausearch_rule_t'
   84 |         time_t sec, unsigned milli, unsigned serial, ausearch_rule_t how);
      |                                                      ^~~~~~~~~~~~~~~
../auparse/auparse.h:85:24: error: expected declaration specifiers or '...' before 'auparse_state_t'
   85 | int ausearch_add_regex(auparse_state_t *au, const char *regexp);
      |                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:86:23: error: expected declaration specifiers or '...' before 'auparse_state_t'
   86 | int ausearch_set_stop(auparse_state_t *au, austop_t where);
      |                       ^~~~~~~~~~~~~~~
../auparse/auparse.h:86:44: error: expected declaration specifiers or '...' before 'austop_t'
   86 | int ausearch_set_stop(auparse_state_t *au, austop_t where);
      |                                            ^~~~~~~~
../auparse/auparse.h:87:21: error: expected declaration specifiers or '...' before 'auparse_state_t'
   87 | void ausearch_clear(auparse_state_t *au);
      |                     ^~~~~~~~~~~~~~~
../auparse/auparse.h:95:23: error: expected declaration specifiers or '...' before 'auparse_state_t'
   95 | int auparse_normalize(auparse_state_t *au, normalize_option_t opt);
      |                       ^~~~~~~~~~~~~~~
../auparse/auparse.h:95:44: error: expected declaration specifiers or '...' before 'normalize_option_t'
   95 | int auparse_normalize(auparse_state_t *au, normalize_option_t opt);
      |                                            ^~~~~~~~~~~~~~~~~~
../auparse/auparse.h:98:52: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
   98 | const char *auparse_normalize_get_event_kind(const auparse_state_t *au);
      |                                                    ^~~~~~~~~~~~~~~
../auparse/auparse.h:98:68: error: expected ';', ',' or ')' before '*' token
   98 | const char *auparse_normalize_get_event_kind(const auparse_state_t *au);
      |                                                                    ^
../auparse/auparse.h:101:31: error: expected declaration specifiers or '...' before 'auparse_state_t'
  101 | int auparse_normalize_session(auparse_state_t *au);
      |                               ^~~~~~~~~~~~~~~
../auparse/auparse.h:104:39: error: expected declaration specifiers or '...' before 'auparse_state_t'
  104 | int auparse_normalize_subject_primary(auparse_state_t *au);
      |                                       ^~~~~~~~~~~~~~~
../auparse/auparse.h:105:41: error: expected declaration specifiers or '...' before 'auparse_state_t'
  105 | int auparse_normalize_subject_secondary(auparse_state_t *au);
      |                                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:106:50: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  106 | const char *auparse_normalize_subject_kind(const auparse_state_t *au);
      |                                                  ^~~~~~~~~~~~~~~
../auparse/auparse.h:106:66: error: expected ';', ',' or ')' before '*' token
  106 | const char *auparse_normalize_subject_kind(const auparse_state_t *au);
      |                                                                  ^
../auparse/auparse.h:107:47: error: expected declaration specifiers or '...' before 'auparse_state_t'
  107 | int auparse_normalize_subject_first_attribute(auparse_state_t *au);
      |                                               ^~~~~~~~~~~~~~~
../auparse/auparse.h:108:46: error: expected declaration specifiers or '...' before 'auparse_state_t'
  108 | int auparse_normalize_subject_next_attribute(auparse_state_t *au);
      |                                              ^~~~~~~~~~~~~~~
../auparse/auparse.h:111:48: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  111 | const char *auparse_normalize_get_action(const auparse_state_t *au);
      |                                                ^~~~~~~~~~~~~~~
../auparse/auparse.h:111:64: error: expected ';', ',' or ')' before '*' token
  111 | const char *auparse_normalize_get_action(const auparse_state_t *au);
      |                                                                ^
../auparse/auparse.h:114:38: error: expected declaration specifiers or '...' before 'auparse_state_t'
  114 | int auparse_normalize_object_primary(auparse_state_t *au);
      |                                      ^~~~~~~~~~~~~~~
../auparse/auparse.h:115:40: error: expected declaration specifiers or '...' before 'auparse_state_t'
  115 | int auparse_normalize_object_secondary(auparse_state_t *au);
      |                                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:116:39: error: expected declaration specifiers or '...' before 'auparse_state_t'
  116 | int auparse_normalize_object_primary2(auparse_state_t *au);
      |                                       ^~~~~~~~~~~~~~~
../auparse/auparse.h:117:46: error: expected declaration specifiers or '...' before 'auparse_state_t'
  117 | int auparse_normalize_object_first_attribute(auparse_state_t *au);
      |                                              ^~~~~~~~~~~~~~~
../auparse/auparse.h:118:45: error: expected declaration specifiers or '...' before 'auparse_state_t'
  118 | int auparse_normalize_object_next_attribute(auparse_state_t *au);
      |                                             ^~~~~~~~~~~~~~~
../auparse/auparse.h:119:49: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  119 | const char *auparse_normalize_object_kind(const auparse_state_t *au);
      |                                                 ^~~~~~~~~~~~~~~
../auparse/auparse.h:119:65: error: expected ';', ',' or ')' before '*' token
  119 | const char *auparse_normalize_object_kind(const auparse_state_t *au);
      |                                                                 ^
../auparse/auparse.h:122:35: error: expected declaration specifiers or '...' before 'auparse_state_t'
  122 | int auparse_normalize_get_results(auparse_state_t *au);
      |                                   ^~~~~~~~~~~~~~~
../auparse/auparse.h:125:41: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  125 | const char *auparse_normalize_how(const auparse_state_t *au);
      |                                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:125:57: error: expected ';', ',' or ')' before '*' token
  125 | const char *auparse_normalize_how(const auparse_state_t *au);
      |                                                         ^
../auparse/auparse.h:128:27: error: expected declaration specifiers or '...' before 'auparse_state_t'
  128 | int auparse_normalize_key(auparse_state_t *au);
      |                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:131:25: error: expected declaration specifiers or '...' before 'auparse_state_t'
  131 | int ausearch_next_event(auparse_state_t *au);
      |                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:132:24: error: expected declaration specifiers or '...' before 'auparse_state_t'
  132 | int ausearch_cur_event(auparse_state_t* au);
      |                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:133:24: error: expected declaration specifiers or '...' before 'auparse_state_t'
  133 | int auparse_next_event(auparse_state_t *au);
      |                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:136:18: error: expected '=', ',', ';', 'asm' or '__attribute__' before '*' token
  136 | const au_event_t *auparse_get_timestamp(const auparse_state_t *au);
      |                  ^
../auparse/auparse.h:137:31: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  137 | time_t auparse_get_time(const auparse_state_t *au);
      |                               ^~~~~~~~~~~~~~~
../auparse/auparse.h:137:47: error: expected ';', ',' or ')' before '*' token
  137 | time_t auparse_get_time(const auparse_state_t *au);
      |                                               ^
../auparse/auparse.h:138:38: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  138 | unsigned int auparse_get_milli(const auparse_state_t *au);
      |                                      ^~~~~~~~~~~~~~~
../auparse/auparse.h:138:54: error: expected ';', ',' or ')' before '*' token
  138 | unsigned int auparse_get_milli(const auparse_state_t *au);
      |                                                      ^
../auparse/auparse.h:139:40: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  139 | unsigned long auparse_get_serial(const auparse_state_t *au);
      |                                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:139:56: error: expected ';', ',' or ')' before '*' token
  139 | unsigned long auparse_get_serial(const auparse_state_t *au);
      |                                                        ^
../auparse/auparse.h:140:36: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  140 | const char *auparse_get_node(const auparse_state_t *au) __attr_dealloc_free;
      |                                    ^~~~~~~~~~~~~~~
../auparse/auparse.h:140:52: error: expected ';', ',' or ')' before '*' token
  140 | const char *auparse_get_node(const auparse_state_t *au) __attr_dealloc_free;
      |                                                    ^
../auparse/auparse.h:141:32: error: type defaults to 'int' in declaration of 'au_event_t' [-Wimplicit-int]
  141 | int auparse_node_compare(const au_event_t *e1, const au_event_t *e2);
      |                                ^~~~~~~~~~
../auparse/auparse.h:141:43: error: expected ';', ',' or ')' before '*' token
  141 | int auparse_node_compare(const au_event_t *e1, const au_event_t *e2);
      |                                           ^
../auparse/auparse.h:142:37: error: type defaults to 'int' in declaration of 'au_event_t' [-Wimplicit-int]
  142 | int auparse_timestamp_compare(const au_event_t *e1, const au_event_t *e2);
      |                                     ^~~~~~~~~~
../auparse/auparse.h:142:48: error: expected ';', ',' or ')' before '*' token
  142 | int auparse_timestamp_compare(const au_event_t *e1, const au_event_t *e2);
      |                                                ^
../auparse/auparse.h:143:44: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  143 | unsigned int auparse_get_num_records(const auparse_state_t *au);
      |                                            ^~~~~~~~~~~~~~~
../auparse/auparse.h:143:60: error: expected ';', ',' or ')' before '*' token
  143 | unsigned int auparse_get_num_records(const auparse_state_t *au);
      |                                                            ^
../auparse/auparse.h:146:26: error: expected declaration specifiers or '...' before 'auparse_state_t'
  146 | int auparse_first_record(auparse_state_t *au);
      |                          ^~~~~~~~~~~~~~~
../auparse/auparse.h:147:25: error: expected declaration specifiers or '...' before 'auparse_state_t'
  147 | int auparse_next_record(auparse_state_t *au);
      |                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:148:43: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  148 | unsigned int auparse_get_record_num(const auparse_state_t *au);
      |                                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:148:59: error: expected ';', ',' or ')' before '*' token
  148 | unsigned int auparse_get_record_num(const auparse_state_t *au);
      |                                                           ^
../auparse/auparse.h:149:29: error: expected declaration specifiers or '...' before 'auparse_state_t'
  149 | int auparse_goto_record_num(auparse_state_t *au, unsigned int num);
      |                             ^~~~~~~~~~~~~~~
../auparse/auparse.h:152:28: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  152 | int auparse_get_type(const auparse_state_t *au);
      |                            ^~~~~~~~~~~~~~~
../auparse/auparse.h:152:44: error: expected ';', ',' or ')' before '*' token
  152 | int auparse_get_type(const auparse_state_t *au);
      |                                            ^
../auparse/auparse.h:153:41: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  153 | const char *auparse_get_type_name(const auparse_state_t *au);
      |                                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:153:57: error: expected ';', ',' or ')' before '*' token
  153 | const char *auparse_get_type_name(const auparse_state_t *au);
      |                                                         ^
../auparse/auparse.h:154:44: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  154 | unsigned int auparse_get_line_number(const auparse_state_t *au);
      |                                            ^~~~~~~~~~~~~~~
../auparse/auparse.h:154:60: error: expected ';', ',' or ')' before '*' token
  154 | unsigned int auparse_get_line_number(const auparse_state_t *au);
      |                                                            ^
../auparse/auparse.h:155:40: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  155 | const char *auparse_get_filename(const auparse_state_t *au);
      |                                        ^~~~~~~~~~~~~~~
../auparse/auparse.h:155:56: error: expected ';', ',' or ')' before '*' token
  155 | const char *auparse_get_filename(const auparse_state_t *au);
      |                                                        ^
../auparse/auparse.h:156:31: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  156 | int auparse_first_field(const auparse_state_t *au);
      |                               ^~~~~~~~~~~~~~~
../auparse/auparse.h:156:47: error: expected ';', ',' or ')' before '*' token
  156 | int auparse_first_field(const auparse_state_t *au);
      |                                               ^
../auparse/auparse.h:157:30: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  157 | int auparse_next_field(const auparse_state_t *au);
      |                              ^~~~~~~~~~~~~~~
../auparse/auparse.h:157:46: error: expected ';', ',' or ')' before '*' token
  157 | int auparse_next_field(const auparse_state_t *au);
      |                                              ^
../auparse/auparse.h:158:43: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  158 | unsigned int auparse_get_num_fields(const auparse_state_t *au);
      |                                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:158:59: error: expected ';', ',' or ')' before '*' token
  158 | unsigned int auparse_get_num_fields(const auparse_state_t *au);
      |                                                           ^
../auparse/auparse.h:159:43: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  159 | const char *auparse_get_record_text(const auparse_state_t *au);
      |                                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:159:59: error: expected ';', ',' or ')' before '*' token
  159 | const char *auparse_get_record_text(const auparse_state_t *au);
      |                                                           ^
../auparse/auparse.h:160:54: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  160 | const char *auparse_get_record_interpretations(const auparse_state_t *au);
      |                                                      ^~~~~~~~~~~~~~~
../auparse/auparse.h:160:70: error: expected ';', ',' or ')' before '*' token
  160 | const char *auparse_get_record_interpretations(const auparse_state_t *au);
      |                                                                      ^
../auparse/auparse.h:161:32: error: expected declaration specifiers or '...' before 'auparse_state_t'
  161 | const char *auparse_find_field(auparse_state_t *au, const char *name);
      |                                ^~~~~~~~~~~~~~~
../auparse/auparse.h:162:37: error: expected declaration specifiers or '...' before 'auparse_state_t'
  162 | const char *auparse_find_field_next(auparse_state_t *au);
      |                                     ^~~~~~~~~~~~~~~
../auparse/auparse.h:163:42: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  163 | unsigned int auparse_get_field_num(const auparse_state_t *au);
      |                                          ^~~~~~~~~~~~~~~
../auparse/auparse.h:163:58: error: expected ';', ',' or ')' before '*' token
  163 | unsigned int auparse_get_field_num(const auparse_state_t *au);
      |                                                          ^
../auparse/auparse.h:164:34: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  164 | int auparse_goto_field_num(const auparse_state_t *au, unsigned int num);
      |                                  ^~~~~~~~~~~~~~~
../auparse/auparse.h:164:50: error: expected ';', ',' or ')' before '*' token
  164 | int auparse_goto_field_num(const auparse_state_t *au, unsigned int num);
      |                                                  ^
../auparse/auparse.h:167:42: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  167 | const char *auparse_get_field_name(const auparse_state_t *au);
      |                                          ^~~~~~~~~~~~~~~
../auparse/auparse.h:167:58: error: expected ';', ',' or ')' before '*' token
  167 | const char *auparse_get_field_name(const auparse_state_t *au);
      |                                                          ^
../auparse/auparse.h:168:41: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  168 | const char *auparse_get_field_str(const auparse_state_t *au);
      |                                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:168:57: error: expected ';', ',' or ')' before '*' token
  168 | const char *auparse_get_field_str(const auparse_state_t *au);
      |                                                         ^
../auparse/auparse.h:169:34: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  169 | int auparse_get_field_type(const auparse_state_t *au);
      |                                  ^~~~~~~~~~~~~~~
../auparse/auparse.h:169:50: error: expected ';', ',' or ')' before '*' token
  169 | int auparse_get_field_type(const auparse_state_t *au);
      |                                                  ^
../auparse/auparse.h:170:33: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  170 | int auparse_get_field_int(const auparse_state_t *au);
      |                                 ^~~~~~~~~~~~~~~
../auparse/auparse.h:170:49: error: expected ';', ',' or ')' before '*' token
  170 | int auparse_get_field_int(const auparse_state_t *au);
      |                                                 ^
../auparse/auparse.h:171:37: error: expected declaration specifiers or '...' before 'auparse_state_t'
  171 | const char *auparse_interpret_field(auparse_state_t *au);
      |                                     ^~~~~~~~~~~~~~~
../auparse/auparse.h:172:46: error: type defaults to 'int' in declaration of 'auparse_state_t' [-Wimplicit-int]
  172 | const char *auparse_interpret_realpath(const auparse_state_t *au);
      |                                              ^~~~~~~~~~~~~~~
../auparse/auparse.h:172:62: error: expected ';', ',' or ')' before '*' token
  172 | const char *auparse_interpret_realpath(const auparse_state_t *au);
      |                                                              ^
../auparse/auparse.h:173:43: error: expected declaration specifiers or '...' before 'auparse_state_t'
  173 | const char *auparse_interpret_sock_family(auparse_state_t *au);
      |                                           ^~~~~~~~~~~~~~~
../auparse/auparse.h:174:41: error: expected declaration specifiers or '...' before 'auparse_state_t'
  174 | const char *auparse_interpret_sock_port(auparse_state_t *au);
      |                                         ^~~~~~~~~~~~~~~
../auparse/auparse.h:175:44: error: expected declaration specifiers or '...' before 'auparse_state_t'
  175 | const char *auparse_interpret_sock_address(auparse_state_t *au);
      |                                            ^~~~~~~~~~~~~~~
auplugin.h:47:37: error: storage class specified for parameter 'auplugin_fgets_state_t'
   47 | typedef struct auplugin_fgets_state auplugin_fgets_state_t;
      |                                     ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:49:1: warning: empty declaration
   49 | enum auplugin_mem {
      | ^~~~
auplugin.h:55:16: error: storage class specified for parameter 'auplugin_callback_ptr'
   55 | typedef void (*auplugin_callback_ptr)(const char *record);
      |                ^~~~~~~~~~~~~~~~~~~~~
auplugin.h:65:29: error: expected declaration specifiers or '...' before 'auplugin_fgets_state_t'
   65 | void auplugin_fgets_destroy(auplugin_fgets_state_t *st);
      |                             ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:66:1: error: expected declaration specifiers before 'auplugin_fgets_state_t'
   66 | auplugin_fgets_state_t *auplugin_fgets_init(void)
      | ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:68:29: error: expected declaration specifiers or '...' before 'auplugin_fgets_state_t'
   68 | void auplugin_fgets_clear_r(auplugin_fgets_state_t *st);
      |                             ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:69:26: error: expected declaration specifiers or '...' before 'auplugin_fgets_state_t'
   69 | int auplugin_fgets_eof_r(auplugin_fgets_state_t *st);
      |                          ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:70:27: error: expected declaration specifiers or '...' before 'auplugin_fgets_state_t'
   70 | int auplugin_fgets_more_r(auplugin_fgets_state_t *st, size_t blen);
      |                           ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:71:22: error: expected declaration specifiers or '...' before 'auplugin_fgets_state_t'
   71 | int auplugin_fgets_r(auplugin_fgets_state_t *st, char *buf, size_t blen, int fd)
      |                      ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:73:24: error: expected declaration specifiers or '...' before 'auplugin_fgets_state_t'
   73 | int auplugin_setvbuf_r(auplugin_fgets_state_t *st, void *buf, size_t buff_size,
      |                        ^~~~~~~~~~~~~~~~~~~~~~
auplugin.h:79:25: error: expected declaration specifiers or '...' before 'auplugin_callback_ptr'
   79 | int auplugin_event_loop(auplugin_callback_ptr callback);
      |                         ^~~~~~~~~~~~~~~~~~~~~
auplugin.h:80:25: error: unknown type name 'auparse_callback_ptr'
   80 | int auplugin_event_feed(auparse_callback_ptr callback);
      |                         ^~~~~~~~~~~~~~~~~~~~
auplugin.c:39:19: error: storage class specified for parameter 'stop'
   39 | static ATOMIC_INT stop = 0;
      |                   ^~~~
auplugin.c:39:1: error: parameter 'stop' is initialized
   39 | static ATOMIC_INT stop = 0;
      | ^~~~~~
auplugin.c:44:12: error: storage class specified for parameter 'fd'
   44 | static int fd;
      |            ^~
auplugin.c:45:18: error: storage class specified for parameter 'outbound_thread'
   45 | static pthread_t outbound_thread;
      |                  ^~~~~~~~~~~~~~~
auplugin.c:48:14: error: storage class specified for parameter 'outbound_thread_loop'
   48 | static void *outbound_thread_loop(void *arg);
      |              ^~~~~~~~~~~~~~~~~~~~
auplugin.c:49:14: error: storage class specified for parameter 'outbound_thread_feed'
   49 | static void *outbound_thread_feed(void *arg);
      |              ^~~~~~~~~~~~~~~~~~~~
auplugin.c:52:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
   52 | {
      | ^
auplugin.c:60:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
   60 | {
      | ^
auplugin.c:64:13: error: storage class specified for parameter 'rx_buf'
   64 | static char rx_buf[MAX_AUDIT_EVENT_FRAME_SIZE+1];
      |             ^~~~~~
auplugin.c:66:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
   66 | {
      | ^
auplugin.c:111:25: error: expected declaration specifiers or '...' before 'auplugin_callback_ptr'
  111 | int auplugin_event_loop(auplugin_callback_ptr callback)
      |                         ^~~~~~~~~~~~~~~~~~~~~
auplugin.c:120:25: error: unknown type name 'auparse_callback_ptr'
  120 | int auplugin_event_feed(auparse_callback_ptr callback)
      |                         ^~~~~~~~~~~~~~~~~~~~
auplugin.c:138:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  138 | {
      | ^
auplugin.c:154:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  154 | {
      | ^
auplugin.c:185:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  185 | {
      | ^
../lib/audit_logging.h:60:7: error: old-style parameter declarations in prototyped function definition
   60 | char *audit_encode_nv_string(const char *name, const char *value,
      |       ^~~~~~~~~~~~~~~~~~~~~~
auplugin.c:217: error: expected '{' at end of input
make[3]: *** [Makefile:544: auplugin.lo] Error 1
make[3]: Leaving directory '/home/grimmauld/coding/audit-userspace/auplugin'
make[2]: *** [Makefile:585: all-recursive] Error 1
make[2]: Leaving directory '/home/grimmauld/coding/audit-userspace/auplugin'
make[1]: *** [Makefile:467: all-recursive] Error 1
make[1]: Leaving directory '/home/grimmauld/coding/audit-userspace'
make: *** [Makefile:399: all] Error 2
```

</p>
</details> 

Bisected to https://github.com/linux-audit/audit-userspace/commit/ae868fa3b19f1f915a1cb53a72d0798a3e8831fa:
```
ae868fa3b19f1f915a1cb53a72d0798a3e8831fa is the first bad commit
commit ae868fa3b19f1f915a1cb53a72d0798a3e8831fa (HEAD)
Author: Steve Grubb <ausearch.1@gmail.com>
Date:   Mon Jun 23 17:48:54 2025 -0400

    Initial commit of the expanded libauplugin

 Makefile.am                             |   2 +-
 audisp/plugins/af_unix/audisp-af_unix.c |   1 -
 auplugin/Makefile.am                    |   5 +++--
 auplugin/auplugin.c                     | 216 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 auplugin/auplugin.h                     |  11 ++++++++++
 common/common.h                         |   1 +
 src/auditd-event.c                      |   1 -
 7 files changed, 232 insertions(+), 5 deletions(-)
 create mode 100644 auplugin/auplugin.c
```

I'll see if i can figure out how to fix this, but no guarantees.

--- a/auparse/auparse.h
+++ b/auparse/auparse.h
@@ -31,6 +31,8 @@
 #endif
 #ifndef __attr_dealloc
 # define __attr_dealloc(dealloc, argno)
+#endif
+#ifndef __attr_dealloc_free
 # define __attr_dealloc_free
 #endif
 #ifndef __attribute_malloc__

--- a/lib/audit_logging.h
+++ b/lib/audit_logging.h
@@ -25,6 +25,7 @@
 
 // Next include is to pick up the function attribute macros
 #include <features.h>
+#include <sys/types.h>
 #include <audit-records.h>
 
 #ifdef __cplusplus

--- a/lib/audit_logging.h
+++ b/lib/audit_logging.h
@@ -40,6 +40,8 @@ extern "C" {
 #endif
 #ifndef __attr_dealloc
 # define __attr_dealloc(dealloc, argno)
+#endif
+#ifndef __attr_dealloc_free
 # define __attr_dealloc_free
 #endif
 // Warn unused result

--- a/lib/libaudit.h
+++ b/lib/libaudit.h
@@ -43,6 +43,8 @@
 // malloc and free assignments
 #ifndef __attr_dealloc
 # define __attr_dealloc(dealloc, argno)
+#endif
+#ifndef __attr_dealloc_free
 # define __attr_dealloc_free
 #endif
 #ifndef __attribute_malloc__

